### PR TITLE
Compaction reshape tasks

### PR DESCRIPTION
**Pull request by Deexie**
Task manager's tasks covering resharding compaction 
on top and shard level.

### CODE REVIEW

**Comment by Deexie** [comment]
@bhalevy review please

**Comment by Deexie** [comment]
- rebase

**Comment by Deexie** [comment]
- rebase

File: compaction/task_manager_module.hh [hunk 1]
```diff
@@ -466,6 +469,87 @@ protected:
...
+
+    virtual std::string type() const override {
+        return "reshaping compaction";
+    }
+protected:
+    virtual future<> run() override = 0;
+};
+
+class reshaping_table_compaction_task_impl : public reshaping_compaction_task_impl {
+private:
```
Comment (raphaelsc) [inline compaction/task_manager_module.hh:494]: for symmetry with others, table_reshape_compaction_task instead? which also makes more sense anyway.

**Comment by Deexie** [comment]
- rename classes

**Review by denesb — APPROVED** [review]

**Pull request merged**